Release: Gateway 2.8.1.2 (#4070)
Kong/docs.konghq.com: app/gateway/changelog.md

no_version: true
---

<!-- vale off -->

## 2.8.1.2
**Release Date** 2022/06/28

### Fixes

#### Enterprise

> * Fixed an issue in hybrid mode where, if a service was set to `enabled: false` and that service had a route with an enabled plugin, any new data planes would receive empty configuration.

Reviewer: ```suggestion
* Fixed an issue in hybrid mode where, if a service was set to `enabled: false` and that service had a route with an enabled plugin, any new data planes would receive an empty configuration.
```

Author: Not sure I agree, since data planes is plural. I don't think that "a configuration" is correct either, so I'm going to leave this one as-is. "Data planes receive configuration from control planes" is a common statement for how Kong Gateway works.

* Fixed a timer leak that occurred when `worker_consistency` was set to `eventual` in `kong.conf`.

> The issue caused timers to be exhausted and failed to start any other timers used by Kong Gateway, showing the error `too many pending timers`.

Reviewer: ```suggestion
This issue caused timers to be exhausted and failed to start any other timers used by Kong Gateway, resulting in a `too many pending timers` error.
```

#### Kong Manager and Dev Portal

* Fixed an issue where Kong Manager did not display all Dev Portal developers in the organization.

> * Fixed an issue with displaying developer role assignment in Kong Manager.

Reviewer: Does this one mean that it just didn't display what roles a developer was assigned to? If so, maybe something like this would work? I think the "displaying" part is throwing me off.
```suggestion
* Fixed an issue that prevented developer role assignments from displaying in Kong Manager.
```

When viewing a role under the Permissions tab in the Dev Portal section, the list of developers wouldn't update when a new developer was added.

> Kong Manager was constructing the wrong URL when retrieving Dev Portal assignees.

Reviewer: ```suggestion
Fixed an issue that caused Kong Manager to construct the wrong URL when it retrieved Dev Portal assignees.
```
Just reworked this to match some of the other "Fixed an issue..." type language and to change the verb tenses.

Author: This isn't a new entry but an explanation for "Fixed an issue that prevented developer role assignments from displaying in Kong Manager." There was an issue, you wouldn't see devs in the list, and it happened because Kong Manager was constructing the wrong URL. So the user wouldn't see the wrong URL, they would see the symptom, which is the missing developers.

Reviewer: That makes sense!

* Fixed empty string handling in Kong Manager. Previously, Kong Manager was handling empty strings as `""` instead of a null value.

> * Improved Kong Manager styling, fixing an issue where content didn't fit on object detail pages.

Reviewer: ```suggestion
* Improved Kong Manager styling by fixing an issue where content didn't fit on object detail pages.
```

> * Fixed an issue with unclickable Kong Manager links and buttons in Safari.

Reviewer: ```suggestion
* Fixed an issue that sometimes prevented clicking Kong Manager links and buttons in Safari.
```
